web/src/js/components/ContentView/ContentEditor.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { Component, PropTypes } from 'react'
+import CodeEditor from '../common/CodeEditor'
+
+export default class ContentEditor extends Component {
+
+    static propTypes = {
+        content: PropTypes.string.isRequired,
+        onSave: PropTypes.func.isRequired,
+        onOpen: PropTypes.func.isRequired,
+        isClosed: PropTypes.bool.isRequired
+    }
+
+    constructor(props){
+        super(props)
+        this.state = {content: this.props.content}
+    }
+
+    render() {
+        return (
+            <div>
+                {this.props.isClosed ?
```

Comment:
This Component is a weird mixup of both the buttons and the code editor. These should be in different components.